evalml/preprocessing/data_splitters/smotenc_split.py

```diff
@@ -0,0 +1,49 @@
+from evalml.preprocessing.data_splitters.base_splitters import (
+    BaseCVSplit,
+    BaseTVSplit
+)
+from evalml.utils import import_or_raise
+
+
+def _allowed_categorical(categorical_features):
```

Comment:
Determines whether the categorical_feature array is a valid input. Since we don't pass the X training data here, we can't do checks on if the length of the input array is longer than the X array itself.